#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: Changed compile time configuration options:
# DP: 
# DP: - Set the default path to comply with Debian policy
# DP: 
# DP: - Define DONT_REPORT_SIGPIPE: We don't want to see `Broken pipe' messages
# DP:   when a job like `cat jobs.c | exit 1' is executed. Fixes part of #7047,
# DP:   #10259, #10433 and #10494. Comment from the upstream author: "The
# DP:   default bash behavior with respect to the exit status of a pipeline
# DP:   will not change.  Changing it as suggested in the discussion of #10494
# DP:   would render bash incompatible with every other shell out there.".
# DP: 
# DP: - Enable System-wide .bashrc file for interactive shells.
# DP: 
# DP: - Enable System-wide .bash_logout  file for interactive shells.
# DP: 
# DP: - make non-interactive shells begun with argv[0][0] == '-'
# DP:   run the startup files when not in posix mode.
# DP: 
# DP: - try to check whether bash is being run by sshd and source
# DP:   the .bashrc if so (like the rshd behavior).

--- bash-2.05a.orig/config-top.h	Wed Oct 24 19:28:49 2001
+++ bash-2.05a/config-top.h	Fri Nov  2 11:16:00 2001
@@ -29,19 +29,19 @@
 
 /* Define DONT_REPORT_SIGPIPE if you don't want to see `Broken pipe' messages
    when a job like `cat jobs.c | exit 1' is executed. */
-/* #define DONT_REPORT_SIGPIPE */
+#define DONT_REPORT_SIGPIPE
 
 /* The default value of the PATH variable. */
 #ifndef DEFAULT_PATH_VALUE
 #define DEFAULT_PATH_VALUE \
-  "/usr/gnu/bin:/usr/local/bin:/usr/ucb:/bin:/usr/bin:."
+  "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
 #endif
 
 /* The value for PATH when invoking `command -p'.  This is only used when
    the Posix.2 confstr () function, or CS_PATH define are not present. */
 #ifndef STANDARD_UTILS_PATH
 #define STANDARD_UTILS_PATH \
-  "/bin:/usr/bin:/usr/ucb:/sbin:/usr/sbin:/etc:/usr/etc"
+  "/bin:/usr/bin:/sbin:/usr/sbin"
 #endif
 
 /* Default primary and secondary prompt strings. */
@@ -49,15 +49,15 @@
 #define SPROMPT "> "
 
 /* System-wide .bashrc file for interactive shells. */
-/* #define SYS_BASHRC "/etc/bash.bashrc" */
+#define SYS_BASHRC "/etc/bash.bashrc"
 
 /* System-wide .bash_logout for login shells. */
-/* #define SYS_BASH_LOGOUT "/etc/bash.bash_logout" */
+#define SYS_BASH_LOGOUT "/etc/bash.bash_logout"
 
 /* Define this to make non-interactive shells begun with argv[0][0] == '-'
    run the startup files when not in posix mode. */
-/* #define NON_INTERACTIVE_LOGIN_SHELLS */
+#define NON_INTERACTIVE_LOGIN_SHELLS
 
 /* Define this if you want bash to try to check whether it's being run by
    sshd and source the .bashrc if so (like the rshd behavior). */
-/* #define SSH_SOURCE_BASHRC */
+#define SSH_SOURCE_BASHRC
